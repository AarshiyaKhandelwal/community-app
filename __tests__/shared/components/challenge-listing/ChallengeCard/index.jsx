import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import ChallengeCard from 'components/challenge-listing/ChallengeCard';
import MockDate from 'mockdate';

beforeAll(() => {
  MockDate.set(1500350400000);
});
afterAll(() => {
  MockDate.reset();
});

const mockData1 = {
  challenge: {
    technologies: 'Data Science',
    allPhases: [{
      phaseType: 'Registration',
      phaseStatus: 'Open',
    }],
    currentPhases: ['Registration'],
    track: 'DEVELOP',
    subTrack: 'CODE',
    events: [{ eventName: 'Submit' }],
    status: 'ACTIVE',
  },
};

const mockData2 = {
  challenge: {
    technologies: 'Go',
    allPhases: [{
      phaseType: 'Submit',
      phaseStatus: 'Open',
    }],
    currentPhases: ['Registration'],
    track: 'DATA_SCIENCE',
    subTrack: 'CODE',
    id: '1',
    totalPrize: 1000,
  },
};

const mockData3 = {
  challenge: {
    technologies: '',
    allPhases: [{
      phaseType: 'Registration',
      phaseStatus: 'Closed',
    }],
    currentPhases: ['Registration'],
    track: 'DATA_SCIENCE',
    subTrack: 'CODE',
    id: 'long id',
    totalPrize: 1000,
  },
};

const mockData4 = {
  challenge: {
    technologies: 'more, technology, to, show',
    allPhases: [{
      phaseType: 'Registration',
      phaseStatus: 'Closed',
    }],
    currentPhases: ['Registration'],
    track: 'DATA_SCIENCE',
    subTrack: 'CODE',
    id: 'long id',
    totalPrize: 1000,
  },
};

const mockData5 = {
  challenge: {
    technologies: '+tag, more, technology, to, show',
    allPhases: [{
      phaseType: 'Registration',
      phaseStatus: 'Closed',
    }],
    currentPhases: ['Registration'],
    track: 'DATA_SCIENCE',
    subTrack: 'CODE',
    id: 'long id',
    totalPrize: 1000,
  },
};

describe('Matches shallow shapshot', () => {
  test('shapshot 1', () => {
    const renderer = new Renderer();

    renderer.render((
      <ChallengeCard {...mockData1} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('shapshot 2', () => {
    const renderer = new Renderer();

    renderer.render((
      <ChallengeCard {...mockData2} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('shapshot 3', () => {
    const renderer = new Renderer();

    renderer.render((
      <ChallengeCard {...mockData3} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <ChallengeCard {...this.props} />;
  }
}

describe('render properly', () => {
  let instance = TU.renderIntoDocument((<Wrapper {...mockData1} />));

  test('another render', () => {
    instance = TU.renderIntoDocument((<Wrapper {...mockData3} />));
  });

  test('click', () => {
    instance = TU.renderIntoDocument((<Wrapper {...mockData4} />));
    const tags = TU.findAllInRenderedTree(instance, item =>
      item && item.className && item.className.match('technology'));
    expect(tags).toHaveLength(4);
    TU.Simulate.click(tags[0]);
  });

  test('click + tag', () => {
    instance = TU.renderIntoDocument((<Wrapper {...mockData5} />));
    const tags = TU.findAllInRenderedTree(instance, item =>
      item && item.className && item.className.match('technology'));
    TU.Simulate.click(tags[0]);
  });
});
