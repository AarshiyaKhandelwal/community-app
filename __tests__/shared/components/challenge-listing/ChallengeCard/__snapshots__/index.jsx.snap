// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot shapshot 1 1`] = `
<div
  className="src-shared-components-challenge-listing-ChallengeCard-___style__challengeCard___1TmHF"
>
  <div
    className="src-shared-components-challenge-listing-ChallengeCard-___style__left-panel___1u9jz"
  >
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-track___3L9Ng"
    >
      <TrackAbbreviationTooltip
        subTrack="CODE"
        track="DEVELOP"
      >
        <span>
          <TrackIcon
            MAIN_URL="https://www.topcoder-dev.com"
            isDataScience={true}
            subTrack="CODE"
            tcoEligible="Submit"
            track="DEVELOP"
          />
        </span>
      </TrackAbbreviationTooltip>
    </div>
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-details___1wy0w src-shared-components-challenge-listing-ChallengeCard-___style__with-register-button___1Gmjf"
    >
      <a
        className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-title___vir_2"
        href="https://www.topcoder-dev.com/challenge-details/undefined/?type=develop"
      />
      <div
        className="src-shared-components-challenge-listing-ChallengeCard-___style__details-footer___3zNR-"
      >
        <span
          className="src-shared-components-challenge-listing-ChallengeCard-___style__date___JV6Mu"
        >
          Ends 
          Jul 18
        </span>
        <Tags
          onTechTagClicked={[Function]}
          technologies="Data Science"
        />
      </div>
    </div>
  </div>
  <div
    className="src-shared-components-challenge-listing-ChallengeCard-___style__right-panel___3kYYO"
  >
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__prizes___3Ms6o src-shared-components-challenge-listing-ChallengeCard-___style__with-register-button___1Gmjf"
    >
      <PrizesTooltip
        challenge={
          Object {
            "allPhases": Array [
              Object {
                "phaseStatus": "Open",
                "phaseType": "Registration",
              },
            ],
            "currentPhases": Array [
              "Registration",
            ],
            "events": Array [
              Object {
                "eventName": "Submit",
              },
            ],
            "isDataScience": true,
            "prize": Array [],
            "status": "ACTIVE",
            "subTrack": "CODE",
            "technologies": "Data Science",
            "track": "DEVELOP",
          }
        }
      >
        <div>
          <div>
            <span
              className="src-shared-components-challenge-listing-ChallengeCard-___style__dollar___8eptG"
            >
              $
            </span>
            0
          </div>
          <div
            className="src-shared-components-challenge-listing-ChallengeCard-___style__label___1gJ8o"
          >
            Purse
          </div>
        </div>
      </PrizesTooltip>
    </div>
    <ChallengeStatus
      challenge={
        Object {
          "allPhases": Array [
            Object {
              "phaseStatus": "Open",
              "phaseType": "Registration",
            },
          ],
          "currentPhases": Array [
            "Registration",
          ],
          "events": Array [
            Object {
              "eventName": "Submit",
            },
          ],
          "isDataScience": true,
          "prize": Array [],
          "status": "ACTIVE",
          "subTrack": "CODE",
          "technologies": "Data Science",
          "track": "DEVELOP",
        }
      }
      detailLink="https://www.topcoder-dev.com/challenge-details/undefined/?type=develop"
      sampleWinnerProfile={undefined}
    />
  </div>
</div>
`;

exports[`Matches shallow shapshot shapshot 2 1`] = `
<div
  className="src-shared-components-challenge-listing-ChallengeCard-___style__challengeCard___1TmHF"
>
  <div
    className="src-shared-components-challenge-listing-ChallengeCard-___style__left-panel___1u9jz"
  >
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-track___3L9Ng"
    >
      <TrackAbbreviationTooltip
        subTrack="CODE"
        track="DATA_SCIENCE"
      >
        <span>
          <TrackIcon
            MAIN_URL="https://www.topcoder-dev.com"
            isDataScience={false}
            subTrack="CODE"
            tcoEligible=""
            track="DATA_SCIENCE"
          />
        </span>
      </TrackAbbreviationTooltip>
    </div>
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-details___1wy0w"
    >
      <a
        className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-title___vir_2"
        href="https://community.topcoder-dev.com/tc?module=MatchDetails&rd=1"
      />
      <div
        className="src-shared-components-challenge-listing-ChallengeCard-___style__details-footer___3zNR-"
      >
        <span
          className="src-shared-components-challenge-listing-ChallengeCard-___style__date___JV6Mu"
        >
          Ended 
          Jul 18
        </span>
        <Tags
          onTechTagClicked={[Function]}
          technologies="Go"
        />
      </div>
    </div>
  </div>
  <div
    className="src-shared-components-challenge-listing-ChallengeCard-___style__right-panel___3kYYO"
  >
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__prizes___3Ms6o"
    >
      <PrizesTooltip
        challenge={
          Object {
            "allPhases": Array [
              Object {
                "phaseStatus": "Open",
                "phaseType": "Submit",
              },
            ],
            "currentPhases": Array [
              "Registration",
            ],
            "id": "1",
            "isDataScience": false,
            "prize": Array [],
            "subTrack": "CODE",
            "technologies": "Go",
            "totalPrize": 1000,
            "track": "DATA_SCIENCE",
          }
        }
      >
        <div>
          <div>
            <span
              className="src-shared-components-challenge-listing-ChallengeCard-___style__dollar___8eptG"
            >
              $
            </span>
            1,000
          </div>
          <div
            className="src-shared-components-challenge-listing-ChallengeCard-___style__label___1gJ8o"
          >
            Purse
          </div>
        </div>
      </PrizesTooltip>
    </div>
    <ChallengeStatus
      challenge={
        Object {
          "allPhases": Array [
            Object {
              "phaseStatus": "Open",
              "phaseType": "Submit",
            },
          ],
          "currentPhases": Array [
            "Registration",
          ],
          "id": "1",
          "isDataScience": false,
          "prize": Array [],
          "subTrack": "CODE",
          "technologies": "Go",
          "totalPrize": 1000,
          "track": "DATA_SCIENCE",
        }
      }
      detailLink="https://community.topcoder-dev.com/tc?module=MatchDetails&rd=1"
      sampleWinnerProfile={undefined}
    />
  </div>
</div>
`;

exports[`Matches shallow shapshot shapshot 3 1`] = `
<div
  className="src-shared-components-challenge-listing-ChallengeCard-___style__challengeCard___1TmHF"
>
  <div
    className="src-shared-components-challenge-listing-ChallengeCard-___style__left-panel___1u9jz"
  >
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-track___3L9Ng"
    >
      <TrackAbbreviationTooltip
        subTrack="CODE"
        track="DATA_SCIENCE"
      >
        <span>
          <TrackIcon
            MAIN_URL="https://www.topcoder-dev.com"
            isDataScience={false}
            subTrack="CODE"
            tcoEligible=""
            track="DATA_SCIENCE"
          />
        </span>
      </TrackAbbreviationTooltip>
    </div>
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-details___1wy0w"
    >
      <a
        className="src-shared-components-challenge-listing-ChallengeCard-___style__challenge-title___vir_2"
        href="https://www.topcoder-dev.com/challenge-details/long id/?type=develop"
      />
      <div
        className="src-shared-components-challenge-listing-ChallengeCard-___style__details-footer___3zNR-"
      >
        <span
          className="src-shared-components-challenge-listing-ChallengeCard-___style__date___JV6Mu"
        >
          Ended 
          Jul 18
        </span>
        <Tags
          onTechTagClicked={[Function]}
          technologies=""
        />
      </div>
    </div>
  </div>
  <div
    className="src-shared-components-challenge-listing-ChallengeCard-___style__right-panel___3kYYO"
  >
    <div
      className="src-shared-components-challenge-listing-ChallengeCard-___style__prizes___3Ms6o"
    >
      <PrizesTooltip
        challenge={
          Object {
            "allPhases": Array [
              Object {
                "phaseStatus": "Closed",
                "phaseType": "Registration",
              },
            ],
            "currentPhases": Array [
              "Registration",
            ],
            "id": "long id",
            "isDataScience": false,
            "prize": Array [],
            "subTrack": "CODE",
            "technologies": "",
            "totalPrize": 1000,
            "track": "DATA_SCIENCE",
          }
        }
      >
        <div>
          <div>
            <span
              className="src-shared-components-challenge-listing-ChallengeCard-___style__dollar___8eptG"
            >
              $
            </span>
            1,000
          </div>
          <div
            className="src-shared-components-challenge-listing-ChallengeCard-___style__label___1gJ8o"
          >
            Purse
          </div>
        </div>
      </PrizesTooltip>
    </div>
    <ChallengeStatus
      challenge={
        Object {
          "allPhases": Array [
            Object {
              "phaseStatus": "Closed",
              "phaseType": "Registration",
            },
          ],
          "currentPhases": Array [
            "Registration",
          ],
          "id": "long id",
          "isDataScience": false,
          "prize": Array [],
          "subTrack": "CODE",
          "technologies": "",
          "totalPrize": 1000,
          "track": "DATA_SCIENCE",
        }
      }
      detailLink="https://www.topcoder-dev.com/challenge-details/long id/?type=develop"
      sampleWinnerProfile={undefined}
    />
  </div>
</div>
`;
